cloudinit/stages.py

```diff
@@ -702,6 +715,34 @@ def _find_networking_config(self):
         return (self.distro.generate_fallback_config(),
                 NetworkConfigSource.fallback)
 
+    def update_event_enabled(
+        self, event_source_type: EventType, scope: EventScope = None
+    ) -> bool:
+        default_events = self.datasource.default_update_events
+        user_events = userdata_to_events(self.cfg.get('updates', {}))
+        # A value in the first will override a value in the second
+        allowed = util.mergemanydict([
+            copy.deepcopy(user_events),
+            copy.deepcopy(default_events),
+        ])
+        LOG.debug('Allowed events: %s', allowed)
+
+        if not scope:
+            scopes = list(allowed.keys())
```

Comment:
Because of the "almost list" structures that dicts give us back.
```
>>> x = {1:2}
>>> [x.keys()]
[dict_keys([1])]
>>> list(x.keys())
[1]
```

Since I'm using a comprehension it later, I need it to be a "real" list.

Reply:
`dict_keys` is iterable:

```py
In [22]: from collections.abc import Iterable
In [21]: isinstance({}.keys(), Iterable)
Out[21]: True
```

So is usable in comprehensions:

```py
In [3]: from enum import Enum

In [4]: class Test(Enum):
   ...:     thing = "string"
In [8]: d = {Test("string"): "foo"}

In [9]: [x for x in d.keys()]
Out[9]: [<Test.thing: 'string'>]

In [10]: [x.value for x in d.keys()]
Out[10]: ['string']
```

It isn't a sequence:

```py
In [22]: from collections.abc import Sequence
In [23]: isinstance({}.keys(), Sequence)
Out[23]: False
```

because it doesn't implement `__getitem__` (so you can't subscript it, which is perhaps what you were thinking of).

Reply:
Actually...no :smile: 

The code I was copying from had a `scopes = [allowed.keys()]` and I misinterpreted why that wasn't working. I thought it was leaving it as a `dict_keys`, which is why I converted it to a proper list, when in actuality it was wrapping the dict_keys into another list.

All this to say...I'll drop the `list(...)`